File: src/main/java/com/jsm/boardgame/web/controller/auth/AuthController.java
package com.jsm.boardgame.web.controller.auth;

import com.jsm.boardgame.service.auth.AuthService;
import com.jsm.boardgame.web.dto.request.auth.LoginRequestDto;
import com.jsm.boardgame.web.dto.request.auth.ReissueRequestDto;
import com.jsm.boardgame.web.dto.response.auth.LoginTokenResponseDto;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RestController
@RequestMapping("/api/v1/auth")
public class AuthController {

    private final AuthService authService;

    @PostMapping("/login")
    public ResponseEntity<LoginTokenResponseDto> login(@Valid @RequestBody LoginRequestDto requestDto) {
        LoginTokenResponseDto responseDto = authService.login(requestDto);
        return ResponseEntity.ok(responseDto);
    }

    @PutMapping("/reissue")
    public ResponseEntity<LoginTokenResponseDto> reissue(@Valid @RequestBody ReissueRequestDto requestDto) {
        LoginTokenResponseDto responseDto = authService.reissue(requestDto);
        return ResponseEntity.ok(responseDto);
    }
}

File: src/main/kotlin/com/jsm/boardgame/auth/presentation/AuthController.kt
package com.jsm.boardgame.auth.presentation

import com.jsm.boardgame.auth.application.dto.request.LoginRequest
import com.jsm.boardgame.auth.application.dto.response.AuthTokenResponse
import com.jsm.boardgame.auth.application.service.AuthService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(
    private val authService: AuthService,
) {

    @PostMapping("/login")
    fun login(@RequestBody request: LoginRequest): AuthTokenResponse =
        authService.login(request.username, request.password)

    @PostMapping("/reissue")
    fun reissue(@RequestBody request: AuthTokenResponse): AuthTokenResponse =
        authService.reissue(request.accessToken, request.refreshToken)
}